posthog/session_recordings/session_recording_api.py

```diff
@@ -614,9 +649,40 @@ def _stream_blob_to_client(
         )
 
         with STREAM_RESPONSE_TO_CLIENT_HISTOGRAM.time():
-            with requests.get(url=url, stream=True) as r:
-                r.raise_for_status()
-                response = HttpResponse(content=r.raw, content_type="application/json")
+            # streams the file from S3 to the client
+            # will not decompress the possibly large file because of `stream=True`
+            #
+            # we pass some headers through to the client
+            # particularly we should signal the content-encoding
+            # to help the client know it needs to decompress
+            #
+            # if the client provides an e-tag we can use it to check if the file has changed
+            # object store will respect this and send back 304 if the file hasn't changed,
+            # and we don't need to send the large file over the wire
+
+            if_none_match = request.headers.get("If-None-Match")
+            headers = {}
+            if if_none_match:
+                headers["If-None-Match"] = ensure_not_weak(if_none_match)
+
+            with stream_from(url=url, headers=headers) as streaming_response:
+                streaming_response.raise_for_status()
+
+                response = HttpResponse(content=streaming_response.raw, status=streaming_response.status_code)
+
+                etag = streaming_response.headers.get("ETag")
+                if etag:
+                    response["ETag"] = ensure_not_weak(etag)
+
+                # blobs are immutable, _really_ we can cache forever
+                # but let's cache for an hour since people won't re-watch too often
+                # we're setting cache control and ETag which might be considered overkill,
+                # but it helps avoid network latency from the client to PostHog, then to object storage, and back again
+                # when a client has a fresh copy
+                response["Cache-Control"] = streaming_response.headers.get("Cache-Control") or "max-age=3600"
+
+                response["Content-Type"] = "application/json"
+
                 response["Content-Disposition"] = "inline"
```

Comment:
content disposition inline is so that the browser doesn't treat the proxied object storage response as a file and download it

interestingly I tried setting the content type to `application/jsonl` and then we _always_ downloaded the file 🤷

Reply:
strangely if I set the content-encoding to "gzip" manually the browser couldn't decode the respone

if I don't set the content-encoding, then the browser displays it as gzip and decodes it successfully 🍤

Reply:
wondering if different browsers could have different behaviors here, probably not, but maybe just double check firefox/chrome/safari?